x/gamm/types/msgs_test.go

```diff
@@ -867,3 +872,148 @@ func TestMsgExitSwapShareAmountIn(t *testing.T) {
 		}
 	}
 }
+
+// Test authz
+func TestAuthzMsg(t *testing.T) {
+	appParams.SetAddressPrefixes()
+	pk1 := ed25519.GenPrivKey().PubKey()
+	addr1 := sdk.AccAddress(pk1.Address()).String()
+	coin := sdk.NewCoin("stake", sdk.NewInt(1))
+	someDate := time.Date(1, 1, 1, 1, 1, 1, 1, time.UTC)
+
+	testCases := []struct {
+		expectedGrantSignByteMsg   string
+		expectedRevokeSignByteMsg  string
+		expectedExecStrSignByteMsg string
+		gammMsg                    sdk.Msg
+	}{
+		{
+			expectedGrantSignByteMsg:   `{"account_number":"1","chain_id":"foo","fee":{"amount":[],"gas":"0"},"memo":"memo","msgs":[{"type":"cosmos-sdk/MsgGrant","value":{"grant":{"authorization":{"type":"cosmos-sdk/GenericAuthorization","value":{"msg":"/osmosis.gamm.v1beta1.MsgExitSwapShareAmountIn"}},"expiration":"0001-01-01T02:01:01.000000001Z"},"grantee":"cosmos1def","granter":"cosmos1abc"}}],"sequence":"1","timeout_height":"1"}`,
```

Comment:
Is there a way to have this represented in a struct format and the unmarshal to json string if needed?

It's hard to follow this in a string format.

Reply:
> Is there a way to have this represented in a struct format and the unmarshal to json string if needed?
> 
> It's hard to follow this in a string format.

I think it's a bit complex. Why we don't use pretty json in expectedGrantSignByteMsg. Like that
```
expectedGrantSignByteMsg :  `{
   "account_number":"1",
   "chain_id":"foo",
   "fee":{
      "amount":[
         
      ],
      "gas":"0"
   },
   "memo":"memo",
   "msgs":[
      {
         "type":"cosmos-sdk/MsgGrant",
         "value":{
            "grant":{
               "authorization":{
                  "type":"cosmos-sdk/GenericAuthorization",
                  "value":{
                     "msg":"/osmosis.gamm.v1beta1.MsgExitSwapShareAmountIn"
                  }
               },
               "expiration":"0001-01-01T02:01:01.000000001Z"
            },
            "grantee":"cosmos1def",
            "granter":"cosmos1abc"
         }
      }
   ],
   "sequence":"1",
   "timeout_height":"1"
}`
```
instead of use one line string

Reply:
Sounds good, thanks for suggesting the alternative